# Using Your Own Data with Terra



Some researchers may choose to bring their own data to Terra:

* To extend or enhance the data currently available from Gen3. Example use cases include:
  * Bringing additional \(e.g. longitudinal\) phenotypic data to enhance the harmonized metadata available from Gen3
  * Joint variant calling with additional researcher provided genomic data
* Or, to use researcher provided data exclusively

At a high level, there are two types of data that researchers typically bring to Terra. Data files \(e.g., genomic data, including CRAM and VCF data\), and metadata \(e.g., tables of clinical/phenotypic or other data, typically regarding the subjects in their study\). These are described separately below.

### Data Files

There are two ways a researcher's data files may be made available in Terra:

1. By uploading data to the researcher's workspace bucket.

* [**Uploading to a workspace Google bucket**](https://support.terra.bio/hc/en-us/articles/360024056512-Uploading-to-a-workspace-Google-bucket)\*\*\*\*

2. By enabling Terra to access the researcher's data in a researcher managed Google bucket.

* \*\*\*\*[**Understanding and setting up a proxy group**](https://support.terra.bio/hc/en-us/articles/360031023592-Understanding-and-setting-up-a-proxy-group)\*\*\*\*

### Metadata

The ways in which a researcher may import metadata to the Terra Data tables are described in these articles.

* \*\*\*\*[**How to import metadata to a workspace data table**](https://support.terra.bio/hc/en-us/articles/360036954991-How-to-import-metadata-to-a-workspace-data-table)\*\*\*\*
* [**Managing data with table**s](https://support.terra.bio/hc/en-us/articles/360025758392)
* _Tutorial on accessing/using dbGaP unharmonized metadata from Gen3 \(coming soon\)_

### Featured Workspace

The following featured workspace provides instruction on bringing your own genomic data to Terra for analysis:

* **TOPMed Alignment and Joint Variant Calling** \(TODO: Link needed\)

